blockonomics-woocommerce.php

```diff
@@ -98,9 +98,35 @@ function woocommerce_handle_blockonomics_return()
         function add_page()
         {
             include_once plugin_dir_path(__FILE__) . 'php' . DIRECTORY_SEPARATOR . 'Blockonomics.php';
+            $blockonomics = new Blockonomics;
+
+            if (isset($_POST['generateSecret']))
+            {
+                generate_secret(true);
+            }
+
+            $api_key = $blockonomics->get_api_key();
             
-            generate_secret();
-            register_setting('blockonomics_g', 'blockonomics_gen_callback', 'gen_callback');
+            // get_api_key() will return api key or temp api key
+            // if both are null, generate new blockonomics guest account with temporary wallet
+            // temp wallet will be used with temp api key
+            if ($api_key == null)
+            {
+                generate_secret();
+                $callback_url = get_callback_url();
+                $response = $blockonomics->get_temp_api_key($callback_url);
+
+                if ($response->response_code != 200)
+                {
+                    $message = __('Error while generating temporary APIKey: '. $response->message, 'blockonomics-bitcoin-payments');
```

Comment:
What if error does not have message? For example 401 or other

Reply:
if get_temp_api_key doesn't work. Its probably a critical error. Very less chance of happening